### DISCUSSION TITLE
Keep getting the following error: Error: [Upstash Redis] The 'url' property is missing or undefined in your Redis config.

### DISCUSSION THREAD

**Discussion by avg-genius** [vercel/community#7343] [Help] [answered]
### Summary

I'm trying to deploy my first project (and I'm new to programming) and I'm having trouble figuring out how to do any of the troubleshooting steps. Can I get someone who can help me out getting this first project deployed? 

### Steps to Reproduce

Error: [Upstash Redis] The 'url' property is missing or undefined in your Redis config.

**Comment by pawlean** [comment]
Hi @avg-genius!

Welcome to the world of programming. 😄 Happy to try and help here! I'm sure that community will also be more than happy to support.

But could you help me understand the issue? Where do you see this error on Vercel?

**Reply by avg-genius** [reply]
![Screenshot 2024-07-02 at 7 58 16 AM](https://github.com/vercel/community/assets/145807944/09dcef93-31c1-47ea-b95b-eec2804c952b)
This error is what I'm getting. I tried again last night and it gave me the same error.

**Comment by avg-genius** [comment]
Well when I go through the steps on Vercel it always stops at the "Build
Logs" steps.

Very Respectfully & Semper Fi,

Bala Selvam
Lead Genius at Average Geniuses
M: 479-445-3166

**Reply by avg-genius** [reply]
Pretty much it doesn't even get past the first step of downloading. I have tried other templates and they deploy just fine.

**Reply by pawlean** [reply]
Can you share the repo you're trying to deploy?

**Reply by avg-genius** [reply]
Hey I actually figured it out! But I'm on to my next bug lol! Here is what I'm seeing and I'm unsure what I'm doing wrong now. Below is the error I'm getting as well as the code that I have. I checked with stackover flow and Open AI and they said it looks good on my end. 
![Screenshot 2024-07-03 at 7 22 43 AM](https://github.com/vercel/community/assets/145807944/37c45992-41c6-4e58-8341-34923aeb26d2)

![Screenshot 2024-07-03 at 7 21 40 AM](https://github.com/vercel/community/assets/145807944/a6746474-5bd3-4704-aae1-875b79e88508)

**Reply by avg-genius** [reply]
@pawlean I also just added you as a collaborator to the repo

**Reply by pawlean** [reply]
@avg-genius It looks like you don't have access to gpt-4o, I would suggest checking your credentials for this in your code.

**Reply by avg-genius** [reply]
Yeah not sure how to check that. I know I have it my OpenAI account but not sure how to add it in the code or how it works. Can you help me figure this out?

**Reply by avg-genius** [reply]
So I've checked everything and according to OpenAI support I do have access to gpt-4o and the API key I have includes gpt-4o

**Reply by pawlean** [reply]
Have you put the API key in your code or in a key/value store?

**Comment by mozeryansky** [comment, marked as answer]
My builds were failing due to: `Error: [Upstash Redis] The 'url' property is missing or undefined in your Redis config.`

The fix was to reintegrate Upstash with Vercel from the Upstash Console.

I had originally setup the integration, and was testing my code locally completely fine. When I pushed my branch, this error happened in the vercel build logs. After doing the integration again it works, and I now see a box for the integration in the Console which wasn't there before.

[a comment by an automated account is not shown]
